src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -512,6 +513,176 @@ pass::EliminateConcat::EliminateConcat() {
+                        return value != 0;
+                    });
+                    if (mask_to_check.empty() || it == mask_to_check.end()) {
+                        return true;
+                    }
+                    return false;
+                };
+                // check that we won't do change dimention rank
+                if (!check_mask(strided_slice_node->get_shrink_axis_mask()) ||
+                    !check_mask(strided_slice_node->get_new_axis_mask()) ||
+                    !check_mask(strided_slice_node->get_ellipsis_mask())) {
+                    return false;
+                }
+
+                auto begin_node = strided_slice_node->get_input_node_shared_ptr(1);
+                const auto& begin_constant_node = ov::util::get_constant_from_source(begin_node);
+                auto begin_values = begin_constant_node->cast_vector<int64_t>();
+
+                auto end_node = strided_slice_node->get_input_node_shared_ptr(2);
+                const auto& end_constant_node = ov::util::get_constant_from_source(end_node);
+                auto end_values = end_constant_node->cast_vector<int64_t>();
+
+                need_opt = true;
+                slice_out_index_in_cocat.push_back(
+                    std::make_tuple(strided_slice_node, begin_values[concat_axis], end_values[concat_axis] - 1));
+            } else {
+                need_opt = false;
+                break;
+            }
+        }
+        if (!need_opt || (slice_out_index_in_cocat.size() == 1))
+            return false;
+
+        uint64_t start_index = 0;
+        std::vector<std::tuple<std::shared_ptr<Node>, int64_t, int64_t>> in_index_in_cocat;
+        for (auto& concat_in : concat_inputs) {
+            auto tmp_index = start_index + concat_in.get_shape()[concat_axis] - 1;
+            in_index_in_cocat.push_back(
+                std::make_tuple(concat_in.get_source_output().get_node_shared_ptr(), start_index, tmp_index));
+            start_index = tmp_index + 1;
+        }
+
+        std::vector<std::tuple<std::shared_ptr<Node>, int64_t, int64_t>> mismatch_slices{};
+        for (const auto& slice_out : slice_out_index_in_cocat) {
+            bool matched = false;
+            for (const auto& concat_in : in_index_in_cocat) {
+                if (get<1>(slice_out) == get<1>(concat_in) && get<2>(slice_out) == get<2>(concat_in)) {
+                    auto slice_outputs = get<0>(slice_out)->outputs();
+                    for (auto& slice_output : slice_outputs) {
+                        replace_output_update_name(slice_output, get<0>(concat_in));
+                    }
+                    matched = true;
+                    break;
+                }
+            }
+            if (!matched)
+                mismatch_slices.push_back(slice_out);
+        }
+
+        int64_t new_start_value{std::numeric_limits<int64_t>::max()};
+        int64_t new_end_value{0};
+        for (const auto& mismatch_slice : mismatch_slices) {
```

Comment:
```suggestion
        for (const auto& [node, begin, end] : mismatch_slices) {
```
Use structured binding to get tuple's elements